src/redux/deviceRedux.tsx

```diff
@@ -0,0 +1,27 @@
+import { Store } from "./initialState";
+import { ActionType } from "./store";
+
+/* selectors */
+export const getDevice = (isDesktop: Store) => isDesktop;
+
+/* action creators */
+export const windowResize = (payload: boolean): DeviceType => ({
+  type: "WINDOW_RESIZE",
+  payload,
+});
+
+//action type
+export interface DeviceType {
+  type: "WINDOW_RESIZE";
+  payload: boolean;
+}
+
+/* reducer */
+export default function reducer(statePart: Store, action: ActionType) {
```

Comment:
```suggestion
export default function reducer(isDesktop: Store["isDesktop"] = true, action: ActionType) {
```

This should also fix the `cannot assign boolean to undefined ` typescript error